File: LordOfTheGames/app/src/main/java/com/example/lordofthegames/Database/CardItemDAO.java
package com.example.progetto.Database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Transaction;

import com.example.progetto.CardItem;

import java.util.List;

@Dao
public interface CardItemDAO {

    //The selected OnConflictStrategy ignores a new CardItem if it's already in the list
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void addCardItem(CardItem cardItem);

    // @Transaction: anything inside the method runs in a single transaction.
    @Transaction
    @Query("SELECT * FROM item ORDER BY item_id DESC")
    LiveData<List<CardItem>> getCardItems();

}

File: LordOfTheGames/app/src/main/java/com/example/lordofthegames/Database/CardItemDAO.kt
package com.example.lordofthegames.Database

import android.view.SurfaceControl.Transaction
import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.lordofthegames.recyclerView.CardItem


@Dao
interface CardItemDAO {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun addCardItem(cardItem: CardItem?)

    fun getCardItems(): LiveData<List<CardItem?>?>?
}

